src/tags/models.py

```diff
@@ -24,7 +24,7 @@ def get_tags_from_user(user):
     BadgeAssertion = apps.get_model("badges", "BadgeAssertion")
 
     # tags related to user's quests
-    quest_ids = QuestSubmission.objects.filter(user=user).distinct().values_list('quest', flat=True)
+    quest_ids = QuestSubmission.objects.filter(user=user, is_approved=True).distinct().values_list('quest', flat=True)
```

Comment:
`QuestSubmission.objects.filter(user=user, is_approved=True)`  Can you check if there is already a QuestSubmission queryset for this? Helps us reduce duplication

Reply:
done. used ```QuestSubmission.objects.all_approve(user=user).distinct()```